src/reshape/index.ts

```diff
@@ -17,13 +17,23 @@ export function reshape<Type, Shape extends Record<string, unknown>>({
 } {
   const result: Record<string, Store<any>> = {};
 
+  const mapUnit = (fn: (value: Type) => any) => (state: Type | null) => {
+    const result = state && fn(state);
+    return result === undefined ? null : result;
+  };
+
   for (const key in shape) {
-    if (key in shape) {
+    if (Object.hasOwnProperty.call(shape, key)) {
       const fn = shape[key];
-      result[key] = source.map((state) => {
-        const result = fn(state);
-        return result === undefined ? null : result;
-      });
+
+      if (is.store(source)) {
+        result[key] = source.map(mapUnit(fn));
+      } else if (is.event(source)) {
+        const tempStore = createStore<Type | null>(null);
+        sample({ source: source, filter: Boolean, target: tempStore });
+
+        result[key] = tempStore.map(mapUnit(fn));
```

Comment:
What do you think, not changing the meaning of the operator, and returning an object of events if passed event? Otherwise, we need to change the type of the value (`| null`) to correctly create a store for each key.

So, I don't understand the case where we need to create stores from the event. We can use `restore`

```ts
const shape = reshape({
  source: restore(eventSource, null),
  shape: {
    length: (original) => original?.length ?? null,
    uppercase: (original) => original?.toUpperCase() ?? null,
    hasSpace: (original) => original?.includes(' ') ?? null,
  },
})
```

Or use any other default value.

I see the case for event -> event mapping for `reshape`:

```ts
const result = reshape({
  source: eventSource,
  shape: {
    length: (original) => original.length,
    uppercase: (original) => original.toUpperCase(),
    hasSpace: (original) => original.includes(' '),
  },
});

is.event(result.length) // true
```

Here we don't have to change the type of the shape (`reshape shape.length` returns `number`, so `result.length` has the same type).